server/broadcast.go

```diff
@@ -81,23 +110,32 @@ func (pi *packetInfo) announceStart() {
 	// If the user has disconnected, then no need to send
 	// the broadcast.
 	if sender == nil {
+		if debugMode {
+			fmt.Printf("[ANNOUNCE] Ignoring message from %s because player no longer exists\n", getIP(pi.conn))
+		}
 		return
 	}
 
-	for _, player := range sender.pool.players {
-		m := message.Signed{
+	announcement := []*message.Signed{
+		{
 			Packet: &message.Packet{
 				Phase:  message.Phase_ANNOUNCEMENT,
 				Number: uint32(pi.tracker.poolSize),
 			},
-		}
+		},
+	}
+
+	for _, player := range sender.pool.players {
```

Comment:
No change in logic but I moved this repeated message creation out of the loop.

Reply:
+1